src/tests/unit/auto/auto_ctput_call_multi.cpp

```diff
@@ -0,0 +1,221 @@
+            });
+        std::vector<std::string> cpuCability{"FP32", "FP16", "INT8", "BIN"};
+        std::vector<std::string> gpuCability{"FP32", "FP16", "BATCHED_BLOB", "BIN", "INT8"};
+        ON_CALL(*core, GetMetric(StrEq(CommonTestUtils::DEVICE_CPU), StrEq(METRIC_KEY(OPTIMIZATION_CAPABILITIES)), _))
+            .WillByDefault(Return(cpuCability));
+        ON_CALL(*core, GetMetric(StrEq(CommonTestUtils::DEVICE_GPU), StrEq(METRIC_KEY(OPTIMIZATION_CAPABILITIES)), _))
+            .WillByDefault(Return(gpuCability));
+
+        ON_CALL(*core,
+                LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                            ::testing::Matcher<const std::string&>(StrEq("CPU")),
+                            ::testing::Matcher<const std::map<std::string, std::string>&>(_)))
+            .WillByDefault(Return(cpuMockExeNetwork));
+
+        ON_CALL(*core,
+                LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                            ::testing::Matcher<const std::string&>(StrEq("GPU")),
+                            ::testing::Matcher<const std::map<std::string, std::string>&>(_)))
+            .WillByDefault(Throw(InferenceEngine::GeneralError{""}));
+        std::shared_ptr<ngraph::Function> simpleNetwork = ngraph::builder::subgraph::makeSingleConv();
+        ASSERT_NO_THROW(simpleCnnNetwork = InferenceEngine::CNNNetwork(simpleNetwork));
+    }
+};
+
+TEST_P(AutoCTPUTCallMulti, CTPUTSingleDevLogicTest) {
+    std::vector<std::string> targetDevices;
+    std::string targetDev;
+    bool AutoCallMulti;
+    Config config;
+    std::tie(AutoCallMulti, targetDevices) = this->GetParam();
+    plugin->SetName("MULTI");
+    for (auto& deviceName : targetDevices) {
+        targetDev += deviceName;
+        targetDev += ((deviceName == targetDevices.back()) ? "" : ",");
+    }
+    if (AutoCallMulti) {
+        config.insert({{CONFIG_KEY(PERFORMANCE_HINT), InferenceEngine::PluginConfigParams::CUMULATIVE_THROUGHPUT}});
+        config.insert({InferenceEngine::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES, targetDev});
+        EXPECT_CALL(*core,
+                    LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                                ::testing::Matcher<const std::string&>(CommonTestUtils::DEVICE_CPU),
+                                ::testing::Matcher<const std::map<std::string, std::string>&>(_)))
+            .Times(1);
+        EXPECT_CALL(*core,
+                    LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                                ::testing::Matcher<const std::string&>(CommonTestUtils::DEVICE_GPU),
+                                ::testing::Matcher<const std::map<std::string, std::string>&>(_)))
+            .Times(1);
+        ASSERT_NO_THROW(plugin->LoadExeNetworkImpl(simpleCnnNetwork, config));
+    } else {
```

Comment:
I don;t see the exe_device test in load fail situation, are u plan to add it?

Reply:
Added it. Please review.